src/postformatting/index.js

```diff
@@ -1,43 +1,50 @@
 import apartment from 'apartment'
-import cssAstFormatter from 'css-fork-pocketjoso'
+import csstree from 'css-tree'
+import debug from 'debug'
 
 import embeddedbase64Remover from './embedded-base64-remover'
 import ffRemover from './unused-fontface-remover'
 import unusedKeyframeRemover from './unused-keyframe-remover'
 
+const debuglog = debug('penthouse:postformatting')
+
 export default function postformatting ({
-  criticalAstRules,
+  astRulesCritical,
   propertiesToRemove,
-  maxEmbeddedBase64Length,
-  debuglog
+  maxEmbeddedBase64Length
 }) {
-  const usedCriticalRules = unusedKeyframeRemover(criticalAstRules)
-  debuglog('postformatting: unusedKeyframeRemover')
+  debuglog('start')
+  const usedCriticalRules = unusedKeyframeRemover(astRulesCritical)
+  debuglog('unusedKeyframeRemover AFTER, length: ' + usedCriticalRules.length)
 
-  let finalCss = cssAstFormatter.stringify({
-    stylesheet: {
-      rules: usedCriticalRules
-    }
+  const finalAst = csstree.fromPlainObject({
+    type: 'StyleSheet',
+    children: usedCriticalRules
   })
-  debuglog('postformatting: stringify from ast')
+  let finalCss = csstree.translate(finalAst)
+  debuglog('stringify from ast')
 
   // remove data-uris that are too long
   // ..faster if this removal can be combined with @font-face one into same iteration..
   finalCss = embeddedbase64Remover(finalCss, maxEmbeddedBase64Length)
-  debuglog('postformatting: embeddedbase64Remover')
+  debuglog('embeddedbase64Remover')
 
   // remove unused @fontface rules
   finalCss = ffRemover(finalCss)
-  debuglog('postformatting: ffRemover')
+  debuglog('ffRemover')
 
-  debuglog('postformatting: propertiesToRemove')
+  debuglog('propertiesToRemove')
   // remove irrelevant css properties
-  finalCss = apartment(finalCss, {
-    properties: propertiesToRemove,
-    // TODO: move into pruneNonCriticalCss script
-    selectors: ['::(-moz-)?selection']
-  })
-  debuglog('postformatting: cleaned css via apartment')
+  try {
+    finalCss = apartment(finalCss, {
```

Comment:
This call to apartment is the only part of Penthouse that still uses the non-fault tolerant rework css parser. Now that the rest of the pipeline is fault tolerant, this call here can be made with css with errors inside, causing rework css’ ast parsing to throw.

Hence my quick fix here is to try/catch and essentially classify this property removal as an "enhancement" for now. This will have the side effect of the critical css increasing in size - only for css files with errors inside. In my tests across thousands of critical css generations This lead to on average about a 0.5% size increase, so nothing dramatic.

Still, this functionality will return. I pinged the author of apartment to give him a chance to say whether he is interested in updating his library, otherwise I'll just rewrite that code and bring it in to penthouse natively - but this would be in another week or so.